src/scene/materials/standard-node-material.js

```diff
@@ -0,0 +1,100 @@
+import { standard } from '../../graphics/program-lib/programs/standard.js';
+
+import {
+    SHADER_FORWARDHDR, SHADER_PICK
+} from '../constants.js';
+
+import { StandardMaterial } from './standard-material.js';
+
+/**
+ * @private
+ * @class
+ * @name pc.StandardNodeMaterial
+ * @augments pc.StandardMaterial
+ * @classdesc StandardNodeMaterial is sub class of the StandardMaterial, and adds shader graph interop functionality
+ * @param {pc.StandardMaterial} mat - optional material which is cloned
+ * @param {any} chunk - optional shader graph node chunk to be used
+ */
+function StandardNodeMaterial(mat, chunk) {
+    StandardMaterial.call(this);
+
+    if (mat) {
+        StandardMaterial.prototype._cloneInternal.call(mat, this);
+    }
+
+    if (chunk) {
+        this._shaderGraphChunk = chunk;
+    }
+}
+
+StandardNodeMaterial.prototype = Object.create(StandardMaterial.prototype);
+StandardNodeMaterial.prototype.constructor = StandardNodeMaterial;
+
+Object.assign(StandardNodeMaterial.prototype, {
+
+    clone: function () {
+        var clone = new StandardNodeMaterial();
+        StandardMaterial.prototype._cloneInternal.call(this, clone);
+
+        clone._shaderGraphChunk = this._shaderGraphChunk;
+
+        return clone;
+    },
+
+    setParameter: function (name, data) {
+        StandardMaterial.prototype.setParameter.call(this, name, data);
+
+        if (this._shaderGraphChunk) {
+            var rootShaderGraph = this._shaderGraphChunk;
+
+            var portName = 'IN_' + name + '_' + rootShaderGraph.id;
+
+            StandardMaterial.prototype.setParameter.call(this, portName, data);
+        }
+    },
+
+    updateUniforms: function () {
+        StandardMaterial.prototype.updateUniforms.call(this);
+
+        if (this._shaderGraphChunk) {
+            var rootShaderGraph = this._shaderGraphChunk;
+
+            rootShaderGraph.updateShaderGraphUniforms(this);
```

Comment:
instead of updateShaderGraphUniforms, maybe call it updateUniforms

Reply:
done - thanks!